instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/EmbeddedConfigFile.java

```diff
@@ -46,20 +46,24 @@ class EmbeddedConfigFile {
   private EmbeddedConfigFile() {}
 
   static OpenTelemetryConfigurationModel extractModel(ConfigurableEnvironment environment) {
-    Map<String, String> props = extractSpringProperties(environment);
+    Map<String, Object> props = extractSpringProperties(environment);
     return convertToOpenTelemetryConfigurationModel(props);
   }
 
-  private static Map<String, String> extractSpringProperties(ConfigurableEnvironment environment) {
+  private static Map<String, Object> extractSpringProperties(ConfigurableEnvironment environment) {
     MutablePropertySources propertySources = environment.getPropertySources();
 
-    Map<String, String> props = new HashMap<>();
+    Map<String, Object> props = new HashMap<>();
     for (PropertySource<?> propertySource : propertySources) {
       if (propertySource instanceof EnumerablePropertySource<?>) {
         for (String propertyName :
             ((EnumerablePropertySource<?>) propertySource).getPropertyNames()) {
           if (propertyName.startsWith("otel.")) {
-            String property = environment.getProperty(propertyName);
+            Object property = propertySource.getProperty(propertyName);
```

Comment:
I needed this change in order to preserve boolean values instead of everything being converted to strings

Reply:
Is this needed for this PR?

Reply:
spring was using type coercion before - because I thought it was necessary.
I'll play around with this PR a little more to get a better picture.

Reply:
I figured it out: the wrong config provider was used: https://github.com/trask/opentelemetry-java-instrumentation/pull/101